packages/federation/src/supergraph.ts

```diff
@@ -717,6 +717,9 @@ export function getStitchedSchemaFromSupergraphSdl(opts: GetSubschemasFromSuperg
     subschemas: [...subschemaMap.values()],
     assumeValid: true,
     assumeValidSDL: true,
+    typeMergingOptions: {
+      useNonNullableFieldOnConflict: true,
+    },
```

Comment:
can you explain a bit this option?

Reply:
Sorry this is the wrong PR. I'll move this commit to the other one

Reply:
This should be the explanation for it.
https://github.com/ardatan/graphql-tools/pull/6091/files#diff-5055db2538d38a1e03e296fb278001444fa669963374ab79723523edf279b8a7